pkg/config/utils.go

```diff
@@ -80,3 +86,21 @@ func WriteConfigToFile(cfg *ClusterConfiguration) error {
 	}
 	return fs.CreateBytes(cfgFilepath, raw)
 }
+
+func SetCurrentEnv(name string) error {
+	envFile := path.Join(viper.GetString("dir"), envFilename)
+	return fs.CreateBytes(envFile, []byte(name))
+}
+
+func GetCurrentEnv() (string, error) {
+	envFile := path.Join(viper.GetString("dir"), envFilename)
+	b, err := ioutil.ReadFile(envFile)
+	if IsNotFound(err) {
+		cName := viper.GetString("cluster-name")
```

Comment:
I think it should be possible to still explicitly set the env (`-c`), even when a default env is set.

Reply:
Alternatively, user configuration files could be implemented (see also #179). Then viper/cobra would take care of precedence automatically.

Reply:
You can still set the environment using `--cluster-name=`. This is just implementing another way of setting it.
We should rename the `cluster-name` flag to `env` to get rid of the confusion.

> Alternatively, user configuration files could be implemented

The `env` option is not environment specific? In my opinion configuration files should be per-environment. This is why I implemented this as just a normal textfile...

Reply:
> You can still set the environment using `--cluster-name=`

`cluster-name` is only considered if the `current-env` file doesn't exist and it is created immediately if it can't be found.

Starting from an empty `/var/lib/kube-spawn`, you can see the problem with e.g.

```
sudo -E ./kube-spawn create -n 2
...
sudo -E ./kube-spawn create -n 3 -c foo
...
sudo -E ./kube-spawn start -c foo
...
spawning cluster "foo" (2 machines)
waiting for machine "kubespawn1" to start up
waiting for machine "kubespawn0" to start up
...
```

That's wrong and 3 nodes should have been spawned, right?

When looking at `destroy`, it looks like this:

```
sudo ./kube-spawn destroy -c foo
using config from /var/lib/kube-spawn/foo
destroying cluster "foo"
force stopping 2 machines
"kubespawn0" stopped
"kubespawn1" stopped
"foo" destroyed
```

Here it partly works: first `default/kspawn.toml` is loaded, but since this is done with viper, `cluster-name` as specified on the CLI is prioritized and the right directory is deleted from disk  (while it stops the wrong containers). The log is confusing since `cfg.Name` is used for that.

https://github.com/kinvolk/kube-spawn/blob/082cb673c4564ea57dc2d5db308a22123561e6ae/pkg/config/utils.go#L62-L65

I would do two things:

* IMHO user configuration should be configurable via cmdline, configuration files and environment variables. For this vanilla viper + cobra could be used, which gives you auto-loading, precedence handling, etc. out of the box. (The fact that viper seems to support only a single configuration file might requires restructuring the current config model a bit, to enable both global (e.g. default env) and per cluster config (e.g. container runtime).)
* Internal configuration can continue to live in `/var/lib/kube-spawn/<env>/kspawn.toml` but should not be loaded with viper (to avoid mixing user and internal configuration), i.e. avoid `viper.Unmarshal(cfg)` and instead populate `ClusterConfiguration` explicitly.